test/addons-napi/test_promise/test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const common = require('../../common');
+const test_promise = require(`./build/${common.buildType}/test_promise`);
+const assert = require('assert');
+
+let expected_result, promise;
+
+// A resolution
+expected_result = 42;
+promise = test_promise.createPromise();
+promise.then(
+  common.mustCall(function(result) {
+    assert.strictEqual(result, expected_result,
+                       'promise resolved as expected');
+  }),
+  common.mustNotCall());
+test_promise.concludeCurrentPromise(expected_result, true);
+
+// A rejection
+expected_result = 'It\'s not you, it\'s me.';
+promise = test_promise.createPromise();
+promise.then(
+  common.mustNotCall(),
+  common.mustCall(function(result) {
+    assert.strictEqual(result, expected_result,
+                       'promise rejected as expected');
+  }));
+test_promise.concludeCurrentPromise(expected_result, false);
```

Comment:
Would be nice to test resolution and rejection behavior with promises as @addaleax noted in https://github.com/nodejs/node/pull/14365#discussion_r134095045.